Sources/Plasma/PubUtilLib/plPipeline/GL/plShaderNode.cpp

```diff
@@ -253,9 +266,25 @@ ST::string plShaderContext::Render()
 
     for (std::shared_ptr<plVaryingNode> node : this->varyings) {
         if (node->count > 1) {
-            out << ST::format("varying {} {}[{}];\n", node->type, node->name, node->count);
+            if(this->type == kVertex) {
+                out << ST::format("out {} {}[{}];\n", node->type, node->name, node->count);
+            } else {
+                out << ST::format("in {} {}[{}];\n", node->type, node->name, node->count);
+            }
+        } else {
+            if(this->type == kVertex) {
+                out << ST::format("out {} {};\n", node->type, node->name);
+            } else {
+                out << ST::format("in {} {};\n", node->type, node->name);
+            }
+        }
+    }
```

Comment:
Automatically inferring if a varying should be in/out. But this conflicts with where I actually need an out in the fragment shader, so I had to declare an explicit out type. Not sure if it would be better to use explicit in/out and get rid of varying, if in/out is a requirement.